charmhelpers/contrib/network/ovs/__init__.py

```diff
@@ -320,25 +325,40 @@ def add_ovsbridge_linuxbridge(name, bridge, ifdata=None):
         ovsbridge_port = "cvo{}".format(base)
         linuxbridge_port = "cvb{}".format(base)
 
+    network_interface_already_exists = False
     interfaces = netifaces.interfaces()
     for interface in interfaces:
         if interface == ovsbridge_port or interface == linuxbridge_port:
             log('Interface {} already exists'.format(interface), level=INFO)
-            return
+            network_interface_already_exists = True
+            break
 
     log('Adding linuxbridge {} to ovsbridge {}'.format(bridge, name),
         level=INFO)
 
-    check_for_eni_source()
+    if not network_interface_already_exists:
+        setup_eni()
+
+        with open('/etc/network/interfaces.d/{}.cfg'.format(
+                linuxbridge_port), 'w') as config:
+            config.write(BRIDGE_TEMPLATE.format(
+                linuxbridge_port=linuxbridge_port,
+                ovsbridge_port=ovsbridge_port, bridge=bridge))
 
-    with open('/etc/network/interfaces.d/{}.cfg'.format(
-            linuxbridge_port), 'w') as config:
-        config.write(BRIDGE_TEMPLATE.format(linuxbridge_port=linuxbridge_port,
-                                            ovsbridge_port=ovsbridge_port,
-                                            bridge=bridge))
+        # NOTE(lourot): 'ifup <name>' can't be replaced by
+        # 'ip link set <name> up' as the latter won't parse
+        # /etc/network/interfaces*
+        ifup_cmd = ['ifup', linuxbridge_port]
+        try:
+            subprocess.check_call(ifup_cmd)
+        except FileNotFoundError:
+            # NOTE(lourot): on bionic and newer, 'ifup' isn't installed by
+            # default
+            apt_install('ifupdown', fatal=True)
+            subprocess.check_call(ifup_cmd)
```

Comment:
I'm fairly against this idea of "automatic" apt install of a library in the c-h code.  This is a dependency that the charm author ought to become aware of and deal with in the pre-install of the charm.

Also, but on bionic what is the alternative to ifup for the user, if it's not installed by default.  Should the code be using that instead as that's the new 'normal' going forwards?

Reply:
Good point. It looks like netplan.io is the new `/etc/network/interfaces`. I'll give it a try.

Reply:
Actually we have decided meanwhile not to support this anymore: https://review.opendev.org/#/c/728382/